ivy/functional/ivy/experimental/manipulation.py

```diff
@@ -30,6 +30,53 @@
 from ivy.utils.exceptions import handle_exceptions
 
 
+@inputs_to_ivy_arrays
+def quantize_linear(x, scale, zero_point, /, *, saturate=False):
+    """
+    Quantizes the input array using the given scale and zero point.
+
+    Parameters
+    ----------
+    x
+        Input array.
+    scale
+        The scale factor.
+    zero_point
+        The zero point offset for quantization.
+    axis
+        The axis along which to quantize. If not set, the array is flattened.
+    saturate
+        Whether to saturate the quantized values to the range [0, 255].
+        Defaults to False.
+
+    Returns
+    -------
+    ret
+        The quantized array.
+
+    Examples
+    --------
+    >>> x = ivy.array([-1.5, 0.3, 2.7, 1.8], dtype=ivy.float32)
+    >>> scale = 0.5
+    >>> zero_point = 128
+    >>> ivy.quantize_linear(x, scale, zero_point)
+    ivy.array([127, 129, 133, 135])
+
+    >>> x = ivy.array([[1.5, 2.7], [0.8, 3.2]], dtype=ivy.float32)
+    >>> scale = 0.1
+    >>> zero_point = 128
+    >>> axis = 1
+    >>> ivy.quantize_linear(x, scale, zero_point, axis=axis)
+    ivy.array([[143., 155.], [136., 160.]])
+    """
+    y = ivy.round(x / scale + zero_point).astype(ivy.uint8)
+
+    if saturate:
+        y = ivy.clip(y, 0, 255)
```

Comment:
based on onnx docs, this function should be generala and not constrained to certain output types or ranges, so I suggest making it more general and remove the hardcoded dtypes and limits

Reply:
There are a couple of specifications when it comes to the dtypes.

Reply:
what is mentioned here is that saturation is dependant on the input dtype, but not strictly restricted to `int8` and `uint8`